File: modules/fbs-core/api/src/main/scala/de/thm/ii/fbs/config/SecurityConfig.scala
package de.thm.ii.fbs.config

import org.springframework.context.annotation.Configuration
import org.springframework.security.web.SecurityFilterChain

import java.util
import javax.servlet.Filter
import javax.servlet.http.HttpServletRequest


@Configuration
class SecurityConfig extends SecurityFilterChain {
  override def matches(request: HttpServletRequest): Boolean = true

  override def getFilters: util.List[Filter] = util.List.of()
}

File: modules/fbs-identity-service/api/src/main/kotlin/de/thm/ii/fbs/fbs_identity_service/config/SecurityConfig.kt
package de.thm.ii.fbs.fbs_identity_service.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.web.SecurityFilterChain

@Configuration
class SecurityConfig {

    @Bean
    fun securityFilterChain(http: HttpSecurity): SecurityFilterChain {
        return http
            .csrf { it.disable() }
            .authorizeHttpRequests {
                it
                    .requestMatchers("/health", "/manifest", "/graphql", "/graphiql", "/api/v1/auth/login", "/error").permitAll()
                    .anyRequest().authenticated()
            }
            .build()
    }
}